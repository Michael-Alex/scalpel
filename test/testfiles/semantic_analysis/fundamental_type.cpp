char               v01;
unsigned char      v02;
signed char        v03;
bool               v04;
unsigned           v05;
unsigned int       v06;
signed             v07;
signed int         v08;
int                v09;
unsigned short int v10;
unsigned short     v11;
unsigned long int  v12;
unsigned long      v13;
signed long int    v14;
signed long        v15;
long int           v16;
long               v17;
signed short int   v18;
signed short       v19;
short int          v20;
short              v21;
wchar_t            v22;
float              v23;
double             v24;
long double        v25;
void*              v26;

