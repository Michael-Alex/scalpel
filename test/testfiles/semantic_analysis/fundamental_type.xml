<namespace id="n0">
 <variable name="v1">
  <type>
   <fundamental_type type="char"/>
  </type>
 </variable>
 <variable name="v2">
  <type>
   <fundamental_type type="unsigned char"/>
  </type>
 </variable>
 <variable name="v3">
  <type>
   <fundamental_type type="signed char"/>
  </type>
 </variable>
 <variable name="v4">
  <type>
   <fundamental_type type="bool"/>
  </type>
 </variable>
 <variable name="v5">
  <type>
   <fundamental_type type="unsigned int"/>
  </type>
 </variable>
 <variable name="v6">
  <type>
   <fundamental_type type="unsigned int"/>
  </type>
 </variable>
 <variable name="v7">
  <type>
   <fundamental_type type="int"/>
  </type>
 </variable>
 <variable name="v8">
  <type>
   <fundamental_type type="int"/>
  </type>
 </variable>
 <variable name="v9">
  <type>
   <fundamental_type type="int"/>
  </type>
 </variable>
 <variable name="v10">
  <type>
   <fundamental_type type="unsigned short int"/>
  </type>
 </variable>
 <variable name="v11">
  <type>
   <fundamental_type type="unsigned short int"/>
  </type>
 </variable>
 <variable name="v12">
  <type>
   <fundamental_type type="unsigned long int"/>
  </type>
 </variable>
 <variable name="v13">
  <type>
   <fundamental_type type="unsigned long int"/>
  </type>
 </variable>
 <variable name="v14">
  <type>
   <fundamental_type type="long int"/>
  </type>
 </variable>
 <variable name="v15">
  <type>
   <fundamental_type type="long int"/>
  </type>
 </variable>
 <variable name="v16">
  <type>
   <fundamental_type type="long int"/>
  </type>
 </variable>
 <variable name="v17">
  <type>
   <fundamental_type type="long int"/>
  </type>
 </variable>
 <variable name="v18">
  <type>
   <fundamental_type type="short int"/>
  </type>
 </variable>
 <variable name="v19">
  <type>
   <fundamental_type type="short int"/>
  </type>
 </variable>
 <variable name="v20">
  <type>
   <fundamental_type type="short int"/>
  </type>
 </variable>
 <variable name="v21">
  <type>
   <fundamental_type type="short int"/>
  </type>
 </variable>
 <variable name="v22">
  <type>
   <fundamental_type type="wchar_t"/>
  </type>
 </variable>
 <variable name="v23">
  <type>
   <fundamental_type type="float"/>
  </type>
 </variable>
 <variable name="v24">
  <type>
   <fundamental_type type="double"/>
  </type>
 </variable>
 <variable name="v25">
  <type>
   <fundamental_type type="long double"/>
  </type>
 </variable>
 <variable name="v26">
  <type>
   <pointer>
    <fundamental_type type="void"/>
   </pointer>
  </type>
 </variable>
</namespace>
