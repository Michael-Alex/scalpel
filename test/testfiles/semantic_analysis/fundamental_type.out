{
  "id": 0,
  "variables":
  [
    {
      "id": 0,
      "name": "v01",
      "type":
      {
        "fundamental type": "char"
      }
    },
    {
      "id": 1,
      "name": "v02",
      "type":
      {
        "fundamental type": "unsigned char"
      }
    },
    {
      "id": 2,
      "name": "v03",
      "type":
      {
        "fundamental type": "signed char"
      }
    },
    {
      "id": 3,
      "name": "v04",
      "type":
      {
        "fundamental type": "bool"
      }
    },
    {
      "id": 4,
      "name": "v05",
      "type":
      {
        "fundamental type": "unsigned int"
      }
    },
    {
      "id": 5,
      "name": "v06",
      "type":
      {
        "fundamental type": "unsigned int"
      }
    },
    {
      "id": 6,
      "name": "v07",
      "type":
      {
        "fundamental type": "int"
      }
    },
    {
      "id": 7,
      "name": "v08",
      "type":
      {
        "fundamental type": "int"
      }
    },
    {
      "id": 8,
      "name": "v09",
      "type":
      {
        "fundamental type": "int"
      }
    },
    {
      "id": 9,
      "name": "v10",
      "type":
      {
        "fundamental type": "unsigned short int"
      }
    },
    {
      "id": 10,
      "name": "v11",
      "type":
      {
        "fundamental type": "unsigned short int"
      }
    },
    {
      "id": 11,
      "name": "v12",
      "type":
      {
        "fundamental type": "unsigned long int"
      }
    },
    {
      "id": 12,
      "name": "v13",
      "type":
      {
        "fundamental type": "unsigned long int"
      }
    },
    {
      "id": 13,
      "name": "v14",
      "type":
      {
        "fundamental type": "long int"
      }
    },
    {
      "id": 14,
      "name": "v15",
      "type":
      {
        "fundamental type": "long int"
      }
    },
    {
      "id": 15,
      "name": "v16",
      "type":
      {
        "fundamental type": "long int"
      }
    },
    {
      "id": 16,
      "name": "v17",
      "type":
      {
        "fundamental type": "long int"
      }
    },
    {
      "id": 17,
      "name": "v18",
      "type":
      {
        "fundamental type": "short int"
      }
    },
    {
      "id": 18,
      "name": "v19",
      "type":
      {
        "fundamental type": "short int"
      }
    },
    {
      "id": 19,
      "name": "v20",
      "type":
      {
        "fundamental type": "short int"
      }
    },
    {
      "id": 20,
      "name": "v21",
      "type":
      {
        "fundamental type": "short int"
      }
    },
    {
      "id": 21,
      "name": "v22",
      "type":
      {
        "fundamental type": "wchar_t"
      }
    },
    {
      "id": 22,
      "name": "v23",
      "type":
      {
        "fundamental type": "float"
      }
    },
    {
      "id": 23,
      "name": "v24",
      "type":
      {
        "fundamental type": "double"
      }
    },
    {
      "id": 24,
      "name": "v25",
      "type":
      {
        "fundamental type": "long double"
      }
    },
    {
      "id": 25,
      "name": "v26",
      "type":
      {
        "pointer":
        {
          "type":
          {
            "fundamental type": "void"
          }
        }
      }
    }
  ]
}
